File: ios/Starter/Adapter/Mapper/AuthorizationMapper.swift
//
//  AuthorizationMapper.swift
//  Starter
//
//  Created by Colton Nohelty on 2/2/19.
//  Copyright © 2019 Colton Nohelty. All rights reserved.
//

import StarterCore
import JWTDecode

class AuthorizationMapper {
    func mapAuthorizationResponse(
        authorizationResponse: AuthorizationResponse,
        accessTokenClaims: JWT,
        refreshTokenClaims: JWT
    ) -> Authorization {
        return Authorization(
            accessToken: authorizationResponse.accessToken,
            accessTokenExpiresAtEpochMilli: accessTokenClaims.expiresAt!.epochMilli,
            accessTokenIssuedAtEpochMilli: accessTokenClaims.issuedAt!.epochMilli,
            tokenType: authorizationResponse.tokenType,
            refreshToken: authorizationResponse.refreshToken,
            refreshTokenExpiresAtEpochMilli: refreshTokenClaims.expiresAt!.epochMilli
        )
    }
}

File: common/src/main/kotlin/com/noheltcj/starter/adapter/mapper/AuthorizationMapper.kt
package com.noheltcj.starter.adapter.mapper

import com.noheltcj.starter.adapter.entity.AuthorizationResponse
import com.noheltcj.starter.adapter.entity.JWTClaims
import com.noheltcj.starter.di.Inject
import com.noheltcj.starter.entity.Authorization

class AuthorizationMapper @Inject constructor() {
    fun mapAuthorizationResponse(
        authorizationResponse: AuthorizationResponse,
        accessTokenClaims: JWTClaims,
        refreshTokenClaims: JWTClaims
    ): Authorization {
        return Authorization(
            accessToken = authorizationResponse.accessToken,
            accessTokenExpiresAtEpochMilli = accessTokenClaims.expiresAtEpochSecond * 1000,
            accessTokenIssuedAtEpochMilli = accessTokenClaims.issuedAtEpochSecond * 1000,
            tokenType = authorizationResponse.tokenType,
            refreshToken = authorizationResponse.refreshToken,
            refreshTokenExpiresAtEpochMilli = refreshTokenClaims.expiresAtEpochSecond * 1000
        )
    }
}